OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -46,11 +49,17 @@ public void RulesetLoaded(Ruleset rules, ActorInfo ai)
 		}
 	}
 
-	public class Explodes : INotifyKilled
+	public class Explodes : INotifyKilled, INotifyDamage
 	{
 		readonly ExplodesInfo info;
 
-		public Explodes(ExplodesInfo info) { this.info = info; }
+		readonly Health health;
+
+		public Explodes(ExplodesInfo info, Actor self)
+		{
+			this.info = info;
+			health = self.Trait<Health>();
```

Comment:
You must declare `Requires<HealthInfo>` on the TraitInfo if you use this from the constructor.